### DISCUSSION TITLE
Allow user to change Mocha timeout

### DISCUSSION THREAD

**Discussion by octaneadmin** [Kong/insomnia#3608] [Ideas]
**Is your feature request related to a problem? Please describe.**
Insomnia Designer currently allows multiple requests to be created as part of a single unit test. This is an important functionality if you have multiple requests relying on a previous ones (request chaining helps, but for more complex situations it fails to satisfy the need).
The problem is that the Mocha timeout is hard-coded to 5000ms. Which means if you add multiple request to a single unit test, you are almost guaranteed to see and error saying the test failed because "Timeout of 5000ms exceeded". 

**Describe the solution you'd like**
Ideally, we either need a way to set the Mocha timeout in User Preferences, or some way to set the timeout within the unit test itself (ex. insomnia.mocha.timeout = 60000). Or if that is a lot of work, then least for the short-term, hard-code the timeout to something much larger (ex. 60 seconds). 

**Describe alternatives you've considered**
Currently we are using postman and we are hoping the whole company can switch to Insomnia Designer if all needed features are present. This has been a roadblock to moving over our more complex API tests and some of our integration tests.

**Additional context**

**Comment by AlexandreLage** [comment]
Importante!

**Comment by andressalro** [comment]
I am presenting the same problem, when executing the unit tests in **Insomnia Designer** there are occasions in which they fail because the necessary execution time of the test exceeds the default time configured in mocha, throwing the following message **"Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done ()" is called; if returning a Promise, ensure it resolves. "**

**On the other hand, in the **Inso** command line of Insomnia the same problem occurs, it is also necessary to include a label to customize the mocha timeout.**

It is a very good tool for testing, but if it is not possible to modify the timeout it greatly limits the great exponential it has.

**Comment by douhaolee** [comment]
yes, me too.
it's a critical feature, I think.
thanks a lot for the good work.

Cheers

**Comment by srmncnk** [comment]
Any progress on this? Or at least hacks?

**Comment by koffie** [comment]
My current workaround is to try to do everything that you would do with multiple requests in one unit test instead[ by chaining the requests ](https://support.insomnia.rest/article/175-chaining-requests#:~:text=Insomnia%20provides%20the%20ability%20to,another%20request%20forms%20a%20link) and then set the trigger behaviour to "When Expired". You might get timeouts on the first run, but it will start reusing results from previous runs on later runs. This of course is only possible for use cases where the reuse of previous requests in the chain is no problem.

**Comment by andressalro** [comment]
something new?

**Comment by andressalro** [comment]
> My current workaround is to try to do everything that you would do with multiple requests in one unit test instead[ by chaining the requests ](https://support.insomnia.rest/article/175-chaining-requests#:~:text=Insomnia%20provides%20the%20ability%20to,another%20request%20forms%20a%20link) and then set the trigger behaviour to "When Expired". You might get timeouts on the first run, but it will start reusing results from previous runs on later runs. This of course is only possible for use cases where the reuse of previous requests in the chain is no problem.

I was also doing it like this, but, I got to a point where I had to chain too many requests so the timeout is inevitable.

I have tried "no history" and "when expired" but if there are three or more chained requests, sometimes, it will execute a double request, in addition to throwing the timeout.

**Comment by koffie** [comment]
P.s. I don't know why this was moved to discussion from an issue. This unconfigurability is actually the single reason that is making me hesitant to adopt insomnia as the end to end testing solution for new projects. I still use it for debugging, but just the idea that there is no real solution might the need for longer tests arise is just really uncomfortable.

**Reply by wdawson** [reply]
Sorry about the sudden change. We are trying to move ideas and feature requests to discussions in general. We're hoping this provides a better experience for the community. We'll still be just as active here and we'll still consider these the same way as we did before even though they're in the new location.

**Comment by isaiahdahl** [comment]
What's the latest on this?

**Comment by wdawson** [comment]
Hey folks! Sorry for missing this one. We extended the timeout to 60 seconds in this PR: #4046 and that was included in the 2021.6.0 release of Insomnia and the 2.4.0 release of inso. Hopefully, that will help everyone run these longer-running tests. 

With that in mind, do folks still need a configurable timeout? Would you want the default to be shorter or longer?

**Reply by xakod** [reply]
What about configurable timeout? 60 seconds cannot be enough sometimes

**Comment by davehouser1** [comment]
Any update on this? Why did they thing expanding the timeout to some static value would be helpful? 
This should be configurable.

**Comment by subnetmarco** [comment]
This should be easy to fix, at the very least it could be set to `60s` for the time being. @filfreire what do you think?

**Reply by subnetmarco** [reply]
My bad - it is already set to `60s`. The solution would be to have this timeout configurable somewhere in the settings.
